branching_xblock/static/js/src/branching_xblock.js

```diff
@@ -4,6 +4,73 @@ function BranchingXBlock(runtime, element) {
     let currentHintNodeId = null;
     let isHintVisible = false;
 
+    const MEDIA_FILE_REGEX = /\.(mp4|webm|ogg|mp3|wav)(\?|#|$)/i;
+
+    function isMediaFile(url) {
+        return MEDIA_FILE_REGEX.test(url || '');
+    }
+
+    function normalizeYouTube(url) {
+        try {
+            const u = new URL(url);
+            const host = u.hostname.toLowerCase();
```

Comment:
@pkulkark, we are doing these two things right now 3 times; we can probably pull it in a sanitizeURL function and do it just one time.